controllers/managedclusterview_controller_test.go

```diff
@@ -0,0 +1,155 @@
+			},
+		}
+	}
+
+	createManagedCluster := func(name, clusterID string) *clusterv1.ManagedCluster {
+		return &clusterv1.ManagedCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: name,
+				Labels: map[string]string{
+					"clusterID": clusterID,
+				},
+			},
+		}
+	}
+
+	t.Run("Create ConfigMap", func(t *testing.T) {
+		mc1 := createManagedCluster("cluster1-name", "cluster1")
+		mc2 := createManagedCluster("cluster2-name", "cluster2")
+		err := c.Create(context.TODO(), mc1)
+		assert.NoError(t, err)
+		err = c.Create(context.TODO(), mc2)
+		assert.NoError(t, err)
+
+		data := map[string]string{
+			"openshift-storage_ocs-storagecluster.config.yaml": `
+                version: "4.Y.Z"
+                deploymentType: "internal"
+                clients:
+                  - name: "client1"
+                    clusterId: "cluster1"
+                  - name: "client2"
+                    clusterId: "cluster2"
+                storageCluster:
+                  namespacedName:
+                    name: "ocs-storagecluster"
+                    namespace: "openshift-storage"
+                  storageProviderEndpoint: ""
+                  cephClusterFSID: "7a3d6b81-a55d-44fe-84d0-46c67cd395ca"
+                storageSystemName: "ocs-storagecluster-storagesystem"
+            `,
+		}
+		mcv := createManagedClusterView("test-view", "local-cluster", data)
+
+		ctx := context.TODO()
+		err = c.Create(ctx, mcv)
+		assert.NoError(t, err)
+
+		err = createOrUpdateConfigMap(ctx, c, *mcv, logger)
+		assert.NoError(t, err)
+
+		cm := &corev1.ConfigMap{}
+		err = c.Get(ctx, types.NamespacedName{Name: ClientInfoConfigMapName, Namespace: os.Getenv("POD_NAMESPACE")}, cm)
+		assert.NoError(t, err)
+		assert.NotNil(t, cm)
+
+		expectedData := map[string]string{
+			"cluster1-name/client1": `{"clusterId":"cluster1","name":"client1","providerInfo":{"version":"4.Y.Z","deploymentType":"internal","storageSystemName":"ocs-storagecluster-storagesystem","providerManagedClusterName":"local-cluster","namespacedName":{"Namespace":"openshift-storage","Name":"ocs-storagecluster"},"storageProviderEndpoint":"","cephClusterFSID":"7a3d6b81-a55d-44fe-84d0-46c67cd395ca"},"clientManagedClusterName":"cluster1-name"}`,
+			"cluster2-name/client2": `{"clusterId":"cluster2","name":"client2","providerInfo":{"version":"4.Y.Z","deploymentType":"internal","storageSystemName":"ocs-storagecluster-storagesystem","providerManagedClusterName":"local-cluster","namespacedName":{"Namespace":"openshift-storage","Name":"ocs-storagecluster"},"storageProviderEndpoint":"","cephClusterFSID":"7a3d6b81-a55d-44fe-84d0-46c67cd395ca"},"clientManagedClusterName":"cluster2-name"}`,
+		}
+
+		assert.Equal(t, expectedData, cm.Data)
+		assert.Equal(t, 1, len(cm.OwnerReferences))
+		assert.Equal(t, mcv.Name, cm.OwnerReferences[0].Name)
+		assert.Equal(t, "ManagedClusterView", cm.OwnerReferences[0].Kind)
+		assert.Equal(t, viewv1beta1.GroupVersion.String(), cm.OwnerReferences[0].APIVersion)
+
+	})
+
```

Comment:
We need another test case here that create MCV in cluster2 namespace, and validates that ConfigMap is updated and has info from both cluster1 and cluster2. The idea of this ConfigMap is to create an aggreagation on all odfinfo configs, and I don't think that is happening right now.

Reply:
Added a testcase for this